Add thrift and tensorpipe backend tests for test_ddp_under_dist_autograd.

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#40210 Add thrift and tensorpipe backend tests for test_ddp_under_dist_autograd.**

Differential Revision: [D22110065](https://our.internmc.facebook.com/intern/diff/D22110065/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D22110065/)!

pritamdamania committed:
Add thrift and tensorpipe backend tests for test_ddp_under_dist_autograd.



Differential Revision: [D22110065](https://our.internmc.facebook.com/intern/diff/D22110065/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D22110065/)!

[ghstack-poisoned]

mrshenli reviewed (commented):
Lint and test failures on this one are real.

osalpekar commented on test/distributed/test_ddp_under_dist_autograd.py:
```diff
@@ -1,596 +1,7 @@
-                    args=(remote_layer1.module_rref, context_id)
-                )
-            )
+from torch.testing._internal.distributed.ddp_under_dist_autograd_test import TestDdpUnderDistAutograd
```
Would a `#noqa` for this line fix the lint error?

osalpekar commented on test/distributed/test_ddp_under_dist_autograd.py:
```diff
@@ -1,596 +1,7 @@
 #!/usr/bin/env python3
```
I think the `run_tests` import needs to be added back to this file to fix the test failures?

pritamdamania committed:
Update on "Add thrift and tensorpipe backend tests for test_ddp_under_dist_autograd."


Differential Revision: [D22110065](https://our.internmc.facebook.com/intern/diff/D22110065/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D22110065/)!

[ghstack-poisoned]

pritamdamania committed:
Update on "Add thrift and tensorpipe backend tests for test_ddp_under_dist_autograd."


Differential Revision: [D22110065](https://our.internmc.facebook.com/intern/diff/D22110065/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D22110065/)!

[ghstack-poisoned]

pritamdamania committed:
Update on "Add thrift and tensorpipe backend tests for test_ddp_under_dist_autograd."


Differential Revision: [D22110065](https://our.internmc.facebook.com/intern/diff/D22110065/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D22110065/)!

[ghstack-poisoned]
